src/hera/shared/_type_util.py

```diff
@@ -0,0 +1,106 @@
+"""Module that handles types and annotations."""
+
+import sys
+from typing import Any, Iterable, List, Optional, Tuple, Type, TypeVar, Union, overload
+
+if sys.version_info >= (3, 9):
+    from typing import Annotated, get_args, get_origin
+else:
+    # Python 3.8 has get_origin() and get_args() but those implementations aren't
+    # Annotated-aware.
+    from typing_extensions import Annotated, get_args, get_origin
+
+if sys.version_info >= (3, 10):
+    from types import UnionType
+else:
+    UnionType = Union
+
+from hera.workflows.artifact import Artifact
+from hera.workflows.parameter import Parameter
+
+
+def is_annotated(annotation: Any):
+    """Check annotation has Annotated type or not."""
+    return get_origin(annotation) is Annotated
+
+
+def unwrap_annotation(annotation: Any) -> Any:
+    """If the given annotation is of type Annotated, return the underlying type, otherwise return the annotation."""
+    if is_annotated(annotation):
+        return get_args(annotation)[0]
+    return annotation
+
+
+T = TypeVar("T")
+V = TypeVar("V")
+
+
+@overload
+def get_annotated_metadata(annotation: Any, type_: Type[T]) -> List[T]: ...
+
+
+@overload
+def get_annotated_metadata(annotation: Any, type_: Tuple[Type[T], Type[V]]) -> List[Union[T, V]]: ...
+
+
+def get_annotated_metadata(annotation, type_):
+    """If given annotation has metadata typed type_, return the metadata.
+
+    Prefer get_workflow_annotatede_metadata if you want to call this with Artifact or Parameter.
```

Comment:
I think you renamed the function 😆 
```suggestion
    Prefer get_workflow_annotation if you want to call this with Artifact or Parameter.
```

Reply:
Oh my you are right. haha thank you!